adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -98,10 +98,11 @@
         sharedSSOCaches.add(msalOAuth2TokenCache);
         mCommonCache = new ADALOAuth2TokenCache(appContext, sharedSSOCaches);
 
-        if (mTokenCacheStore instanceof DefaultTokenCacheStore) {
+        if (mTokenCacheStore instanceof DelegatingCache) {
+            final ITokenCacheStore delegate = ((DelegatingCache) mTokenCacheStore).getDelegateCache();
             //If the default token cache is in use... delegate token operations to unified cache in common
             //If not using default token cache then sharing SSO state between ADAL & MSAL cache implementations will not be possible anyway
-            mUseCommonCache = true;
+            mUseCommonCache = delegate instanceof DefaultTokenCacheStore;
```

Comment:
Consider using `isAssignableFrom`

Reply:
Is it possible to write a unit test here that highlights the problem, and then is fixed by this code change?

Reply:
21b0ce12526b95bb6a803b75b059282e31c6272e